test/Interop/Cxx/class/noncopyable-irgen.swift

```diff
@@ -0,0 +1,113 @@
+// RUN: %target-swift-frontend -cxx-interoperability-mode=default -emit-ir -I %swift_src_root/lib/ClangImporter/SwiftBridging -I %t/Inputs %t/test.swift -Xcc -fignore-exceptions -verify -verify-additional-file %t/Inputs/noncopyable.h -verify-additional-prefix TEST3- -D TEST3
+// RUN: %target-swift-frontend -cxx-interoperability-mode=default -emit-ir -I %swift_src_root/lib/ClangImporter/SwiftBridging -I %t/Inputs %t/test.swift -Xcc -fignore-exceptions -verify -verify-additional-file %t/Inputs/noncopyable.h -verify-additional-prefix TEST4- -D TEST4 -Xcc -std=c++20
+
+//--- Inputs/module.modulemap
+module Test {
+    header "noncopyable.h"
+    requires cplusplus
+}
+
+//--- Inputs/noncopyable.h
+#include "swift/bridging"
+#include <string>
+
+struct NonCopyable {
+    NonCopyable() = default;
+    NonCopyable(const NonCopyable& other) = delete; // expected-note {{'NonCopyable' has been explicitly marked deleted here}}
+    NonCopyable(NonCopyable&& other) = default;
+};
+
+template <typename T>
+struct OwnsT {
+    T element;
+    OwnsT() {}
+    OwnsT(const OwnsT &other) : element(other.element) {} 
+    // expected-error@-1 *{{call to deleted constructor of 'NonCopyable'}}
+    // expected-note@-2 *{{in instantiation of member function 'OwnsT<NonCopyable>::OwnsT' requested here}}
+    // expected-TEST1-error@-3 {{failed to copy 'OwnsT<NonCopyable>'; did you want to import 'OwnsT<NonCopyable>' as ~Copyable?}}
+    // expected-TEST1-note@-4 {{use 'requires' (since C++20) to specify the constraints under which the copy constructor is available}}
+    // expected-TEST1-note@-5 {{annotate a type with 'SWIFT_COPYABLE_IF(<T>)' in C++ to specify that the type is Copyable if <T> is Copyable}}
+    // expected-TEST1-note@-6 {{annotate a type with 'SWIFT_NONCOPYABLE' in C++ to import it as ~Copyable}}
+    OwnsT(OwnsT&& other) {}
+};
+
+using OwnsNonCopyable = OwnsT<NonCopyable>;
+
+template <typename T> struct Derived : OwnsT<T> {
+    // expected-TEST2-error@-1 {{failed to copy 'Derived<NonCopyable>'; did you want to import 'Derived<NonCopyable>' as ~Copyable?}}
+    // expected-TEST2-note@-2 {{use 'requires' (since C++20) to specify the constraints under which the copy constructor is available}}
+    // expected-TEST2-note@-3 {{annotate a type with 'SWIFT_COPYABLE_IF(<T>)' in C++ to specify that the type is Copyable if <T> is Copyable}}
+    // expected-TEST2-note@-4 {{annotate a type with 'SWIFT_NONCOPYABLE' in C++ to import it as ~Copyable}}
+};
+
+using DerivedNonCopyable = Derived<NonCopyable>;
+
+template <typename T> struct SWIFT_COPYABLE_IF(T) Annotated {
+    T element;
+    Annotated() : element() {}
+    Annotated(const Annotated &other) : element(other.element) {}
+    // expected-TEST3-error@-1 {{failed to copy 'Annotated<OwnsT<NonCopyable>>'; did you want to import 'Annotated<OwnsT<NonCopyable>>' as ~Copyable?}}
```

Comment:
Perhaps "mean" or "intend" instead of "want"? I haven't looked at what other diagnostics say though, we should follow whatever pattern they use